react-components/src/components/RevealTopbar/RuleBasedOutputsContainer.tsx

```diff
@@ -0,0 +1,103 @@
+/*!
+ * Copyright 2024 Cognite AS
+ */
+import { useState, type ReactElement, useEffect, useMemo } from 'react';
+
+import { Button, Dropdown, Flex, Icon, Menu, Switch } from '@cognite/cogs.js';
+
+import { RuleBasedOutputs } from '../RuleBasedOutputs/RuleBasedOutputs';
+import { RULE_BASED_OUTPUTS_VIEW } from '../RuleBasedOutputs/constants';
+import { FdmSDK } from '../../utilities/FdmSDK';
+import { useSDK } from '../RevealCanvas/SDKProvider';
+import { type RuleAndEnabled } from '../RuleBasedOutputs/types';
+import { useTranslation } from '../i18n/I18n';
+
+export const RuleBasedOutputsContainer = (): ReactElement => {
+  const [ruleInstances, setRuleInstances] = useState<RuleAndEnabled[]>();
+  const [currentRuleSetEnabled, setCurrentRuleSetEnabled] = useState<any>();
+  const sdk = useSDK();
+  const fdmSdk = useMemo(() => new FdmSDK(sdk), [sdk]);
+
+  const { t } = useTranslation();
+
+  useEffect(() => {
```

Comment:
This should be a `useQuery` the only difference will be that instead of the `ruleInstances ... = useState...`, we can write `const ruleInstancesResult = useQuery(['react-components', 'color-overlay-rules', 'all'], <async-function>);` where the `<async-function>` is the same as the one you wrote here, but returns the value instead of using the `setRuleInstances`. Then, the result will be put in `ruleInstancesResult.data` and triggering a React re-render when it's ready. Let me know if you want help with this

Reply:
Refactored here:
https://github.com/cognitedata/reveal/pull/4210/commits/e4c1637a2c6f4412f8fa79861d3d37844a646860